server/controllers/profile.js

```diff
@@ -38,4 +36,35 @@ exports.loadProfile = asyncHandler(async (req, res, next) => {
       profile: profile,
     },
   });
+});
+
+// @route GET /profile/sitter
+// @desc Get all profile data
+// @access Private
+exports.getAllSitters = asyncHandler(async (req, res) => {
+  const result = [];
+  const profiles = await Profile.find({ isSitter: true, price: { $ne: null } })
+  if (profiles) {
+    profiles.filter((profile) => {
+      const address = profile.address;
+      if (address.toLowerCase().search(req.query.location.toLowerCase()) > -1) {
+        result.push(profile);
+      }
+    })
+    if (result.length) {
+      res.status(200).json({
+        success: {
+          profiles: result
+        },
+      });
+    }
+    else {
+      res.status(204).json({});
+      throw new Error("No Profile found");
+    }
```

Comment:
This is not the appropriate status code, but the block is redundant anyway.

Reply:
Removed the else clause